src/workflow/ParameterManager.py

```
@@ -287,4 +288,67 @@ def clear_parameter_session_state(self) -> None:
+            # OpenMS INI files usually encapsulate everything in a top-level <NODE name="ToolName">.
+            # We must strip this prefix so the keys perfectly match the JSON session state keys.
+            tool_prefix = f"{tool}:1:"
+            cleaned_params = [
+                p[len(tool_prefix):] if p.startswith(tool_prefix) else p 
+                for p in bool_params
+            ]
+
+            return cleaned_params
+
+        except ET.ParseError as e: 
+            logger.error(f"XML parsing failed for {tool}: {e}"); raise
```

Ruff (0.15.5)

[error] Undefined name `logger`

(F821)